### ISSUE TITLE
firstBoxerRecord not available on boxRec.getDate

### ISSUE THREAD

**Issue by rubengmurray** [boxing/boxrec#213]
When requesting information from `boxRec.getDate` and looping the individual events then the individual bouts, the firstBoxerRecords for bouts in the future are always blank...

Has this always been the case / a known issue?

It appears this is because it's being sourced from this page: `https://boxrec.com/en/proboxer/565085?toggleRatings=y`, where the firstBoxerRecord (profile page) is in an entirely different area to the secondBoxerRecords (opponents in the table).

Re-producible with the following: 

```
const cookieJar = await boxrec.login('your_key', 'your_pass')
const response = await boxrec.getDate(cookieJar, dateToGet)
response.events.forEach(event => {
  event.bouts.forEach(bout => {
    console.log(bout.firstBoxerRecord)
  }
}
```

Never used cheerio before so not sure how to fix easily.

**Comment by mikedidomizio** [comment]
Hey,
Sorry forgot to reply to this.

I didn't seem to get this issue and it seems to be working fine, although I'm a bit confused with what you have commented here.

If you do a `getDate` which you have, it goes through to `https://boxrec.com/en/date`.  I'm not sure where you're feeling it is sourced from `proboxer/565085`.

The `firstBoxerRecord` on the date/event/scheduled pages is _their record going into that bout_ and not their most up to date current record.

**Comment by rubengmurray** [comment]
Hi Mike,

I've just tested this again now - seeing the same as I was previously:

````
const getEventsByDate = async () => {
  const cookieJar = await boxrec.login('your_key', 'your_pass')
  const response = await boxrec.getDate(cookieJar, '2019-10-08')
  response.events.forEach(event => {
    event.bouts.forEach(bout => {
      console.log(bout.firstBoxerRecord) // always returns { draw: null, loss: null, win: null }
      console.log(bout.secondBoxerRecord) // always returns valid integers on draw, loss & win
    }
  }
}

getEventsByDate()
````

My point to the code was that it looks like the `getDate` function sources it's data for bouts from the same place that the `getEventById` does - as it's the same response structure. I had a look in the code locally and could see a function called `firstBoxerRecord()` that pulls it's data from a column that's close to the same place `secondBoxerRecord()` pulls it's data from - successfully - even though it's not in the same place on that page.

Hope that helps shed some light?

````
const getEventByDate = async () => {
  const cookieJar = await boxrec.login('your_key', 'your_pass')
  const event= await boxrec.eventById(cookieJar, 796410)
    event.bouts.forEach(bout => {
      console.log(bout.firstBoxerRecord) // always returns { draw: null, loss: null, win: null }
      console.log(bout.secondBoxerRecord) // always returns valid integers on draw, loss & win
  }
````

**Comment by mikedidomizio** [comment]
Yeah so both use a similar structure for consistency which is [this object "Record"](https://boxrec-npm-docs.s3.us-east-2.amazonaws.com/master/interfaces/record.html).  Any place records are returned will use this structure.

It is possible for `win/loss/draw` to be `null` _if_ the fighter has an unknown record (very obscure fighter) or the scheduled fighter is unknown (no one selected to fight).  Is this possibly what you are seeing?  If you pick a future date you will probably encounter a few `null` values.

In terms of "2019-10-08", I used your code above for `getDate` and it seems fine.  I added in the `firstBoxer` for display purposes.

My code:
```
console.log(bout.firstBoxer);
console.log(bout.firstBoxerRecord); // always returns { draw: null, loss: null, win: null }
console.log(bout.secondBoxerRecord);
```

output:
```
{ id: 866651, name: 'Chol Mi Pang' }       
{ draw: 0, loss: 0, win: 6 }               
{ draw: 0, loss: 3, win: 4 }               
{ id: 899631, name: 'Tsukimi Namiki' }     
{ draw: 0, loss: 1, win: 5 }               
{ draw: 0, loss: 3, win: 1 }               
{ id: 889134, name: 'Yu Ting Lin' }        
{ draw: 0, loss: 0, win: 5 }               
{ draw: 0, loss: 1, win: 1 }               
{ id: 899670, name: 'Nilawan Techaseup' }  
{ draw: 0, loss: 1, win: 2 }               
{ draw: 0, loss: 0, win: 2 }               
```

BoxRec:
![image](https://user-images.githubusercontent.com/5728044/66498036-6b1e1c80-ea8b-11e9-80c6-6c46266d6802.png)

**Comment by rubengmurray** [comment]
Hi Mike,

I get flat out `null` for every key on the `firstBoxerRecord` object: `win`, `draw` and `loss`, for dates '2019-10-10', '2019-10-11', and even '2019-10-12' - where I'd expect for Saturday at least there to be some data?

In fact, I'd never checked a date in the past before, and I am seeing data on all of these keys for events in the past. Strange...

Could this be down to an API key setting or something?

**Comment by mikedidomizio** [comment]
Okay, I misread before and I thought it was only the first row of `firstBoxerRecord` that was `null`, it seems all values are `null` for `firstBoxerRecord`.

Can you go to BoxRec, log into BoxRec, go to a `date` page with table rows and enter this into developer console:

`$("#calendarDate thead tr th").length`

What do you get?
I get "13", maybe column numbers are different for you?

Because I'm stuck understanding the issue, let's try some things:
1.  Ensure up to date with version `6.2.0`
2.  (Optionally) Create a new user on BoxRec if possible
3.  Make the request, and verify the values match on the associated BoxRec page

There is no API key setting, so it works for no one or it works for everyone

**Comment by rubengmurray** [comment]
I'll try and get round to verifying the above this weekend.

**Comment by rubengmurray** [comment]
Hi Mike,

I ran the `.length` on a date page in the future and it came out at 13 as well. 

https://boxrec.com/en/date?date=2019-10-26

Also definitely on `6.2.0`.

But, I've just had a look to see if I can find any differences between a date page in the future and a date page in the past and there is one... the `.length` of pages in the past is 16. It contains the lbs of the fighter (plus one other phantom column). I'm seeing records for bouts in the past that I wasn't previously when they were in the future, so I think this must be related to the issue...

 https://boxrec.com/en/date?date=2019-10-19

I've had a look in the code but not sure I can make it out once it gets to those `_awaiter` and `_generator` functions inside `boxrec-requests` without a fair bit of effort. Let me know your thoughts.

Cheers.

**Comment by mikedidomizio** [comment]
That looks to be the issue.  I'll create an issue for this

**Comment by mikedidomizio** [comment]
https://github.com/boxing/boxrec/issues/219

**Comment by mikedidomizio** [comment]
This should be fixed in version [7.0.1](https://github.com/boxing/boxrec/pull/224).  Update to 7.0.1 and let me know if that works for you

**Comment by rubengmurray** [comment]
Confirmed as working on 7.0.1. Thanks for the fix Mike.
